### PR TITLE
Pipeline, Kapitolina Untila - 21FPL1

### PR DESCRIPTION
**Pull request by KapitolinaVU**

### CODE REVIEW

File: lab_6_pipeline/pipeline.py [hunk 1]
```diff
@@ -3,12 +3,41 @@
...
+from core_utils.article.article import SentenceProtocol, get_article_id_from_filepath, split_by_sentence
+from core_utils.article.io import from_raw, to_cleaned
 from core_utils.article.ud import OpencorporaTagProtocol, TagConverter
+from core_utils.constants import ASSETS_PATH
 
 
 # pylint: disable=too-few-public-methods
+
+
+class FileNotFoundError(Exception):
```
Comment (marina-kaz) [inline lab_6_pipeline/pipeline.py]: This exception exists already

File: lab_6_pipeline/pipeline.py [hunk 2]
```diff
@@ -3,12 +3,41 @@
...
+    """
+    file does not exist
+    """
+
+
+class NotADirectoryError(Exception):
```
Comment (marina-kaz) [inline lab_6_pipeline/pipeline.py]: This exception exists already

File: lab_6_pipeline/pipeline.py [hunk 3]
```diff
@@ -18,21 +47,48 @@ def __init__(self, path_to_raw_txt_data: Path):
...
+            raise EmptyDirectoryError
+
+        list_of_ids = []
+        for file in self.path_to_raw_txt_data.glob("*_raw.txt"):
+            print(file)
+            if not file.stat().st_size:
+                raise InconsistentDatasetError
+
+            list_of_ids.append(int(file.name[:file.name.index('_')]))
+            if sorted(list_of_ids) != list(range(1, len(list_of_ids) + 1)):
```
Comment (marina-kaz) [inline lab_6_pipeline/pipeline.py]: Is it necessary to check the order after every single iteration? Could it be done all at once after all the IDs are collected?

File: lab_6_pipeline/pipeline.py [hunk 4]
```diff
@@ -144,16 +213,27 @@ def __init__(self, corpus_manager: CorpusManager):
...
+        self._corpus = corpus_manager
 
     def _process(self, text: str) -> List[ConlluSentence]:
         """
         Returns the text representation as the list of ConlluSentence
         """
+        conllusentences = []
+        sentences = split_by_sentence(text)
+        for num, sentence in enumerate(sentences):
+            tokens = [ConlluToken(word.strip()) for word in sentence.split()]
```
Comment (marina-kaz) [inline lab_6_pipeline/pipeline.py]: Please, explain why you need to strip the `word`

**Review by marina-kaz — COMMENTED** [review]
Looks good for the mark 4!
Maybe try for a better mark?

File: lab_6_pipeline/pipeline.py [hunk 5]
```diff
@@ -144,16 +257,68 @@ def __init__(self, corpus_manager: CorpusManager):
...
         """
+        conllu_sentences = []
+        result = (i for i in self._mystem.analyze(text))
+
+        for idx_sent, sentence in enumerate(split_by_sentence(text)):
+            conllu_tokens = []
+            tokens = []
+
+            for token in result:
+                if token['text'] not in sentence:
```
Comment (marina-kaz) [inline lab_6_pipeline/pipeline.py:277]: can you explain in which cases this condition is true?

**Review by marina-kaz — COMMENTED** [review]
Good job!
Be ready to explain how the pipeline works.

File: lab_6_pipeline/pipeline.py (hunk 5 above)
Comment (KapitolinaVU) [reply lab_6_pipeline/pipeline.py:277]: it is true when the word (from analyzed by mystem text) is not in current sentence

**Pull request closed without merging**